backend/internal/system/cert/cert.go

```diff
@@ -53,3 +56,28 @@ func GetTLSConfig(cfg *config.Config, currentDirectory string) (*tls.Config, err
 		MinVersion:   tls.VersionTLS12, // Enforce minimum TLS version 1.2
 	}, nil
 }
+
+// GetCertificateKid extracts the Key ID (kid) from the TLS certificate using SHA-256 thumbprint.
+func GetCertificateKid(cfg *config.Config, currentDirectory string) (string, error) {
```

Comment:
No need to pass the thunder runtime as arguments. It can be retrieved from any code